Sources/SwiftParser/Modifiers.swift

```diff
@@ -84,7 +84,8 @@ extension Parser {
         (.declarationModifier(._const), let handle)?,
         (.declarationModifier(._local), let handle)?,
         (.declarationModifier(.__setter_access), let handle)?,
-        (.declarationModifier(.reasync), let handle)?:
+        (.declarationModifier(.reasync), let handle)?,
+        (.declarationModifier(.resultDependsOnSelf), let handle)?:
```

Comment:
We should only parse `resultDependsOnSelf` if the experimental language features is enabled. I.e. this something like

```swift
case (.declarationModifier(.resultDependsOnSelf), let handle)? where experimentalFeatures.contains(. nonEscapableTypes)
```

It would also be good to have test case that `resultDependsOnSelf` does not parse if the experimental language feature is not enabled.